docs/migration-guide.rst

```diff
@@ -0,0 +1,56 @@
+.. title:: Ouster-ROS Driver Migration Guide
+
+
+===================================
+Migrating from 20220608 to 20220826
+===================================
+The ``20220826`` release brought several improvements to the ``ouster-ros``, however, these
+improvements included several breaking changes. This guide summarizes these change and how to
+mitigate each one.
+
+Launch Files:
+=============
+The ``ouster.launch`` has been split into three functionally different launch files:
+``sensor.launch``, ``replay.launch``, and ``record.launch``. Each of these launch files only accepts
+parameters that are relevant to their functional use. For example, the ``sensor_hostname`` argument
+is only valid and required when using the ``sensor.launch`` and ``record.launch`` launch files.
+The ``metadata`` argument however is no longer required when connecting to the sensor in live mode.
+It is still required when using ``record.launch`` and ``replay.launch``. Refer to the `main
+documentation <./doc/index.rst>`_ for more details on the different cases of usage.
+
+Topics:
+=======
+Rather than having each topic published by ``ouster-ros`` be prefixed with the name of the ros node
+that publishes it, the topic names of all ros nodes that compose the ``ouster-ros`` driver have been
+combined under a single namespace. Thus all topics would appear prefixed by the new namespace.
+
+If a user wishes to maintain the old topic names then they can achieve that by utilizing the
+``<remap>`` tag in ros launch files. For example, let's say we want to remap the three topics
+published by ``os_cloud_node`` to their old names when connecting to a sensor through the
+``sensor.launch`` file. To do so, we need to edit the ``sensor.launch`` file and add the three
+following remap tags right before any node is instantiated (i.e. the ``<remap>`` tag defintions
+should precede any ``<node>`` tags in the launch file)::
+
+    <remap from="/$(arg ouster_ns)/imu" to="/os_cloud_node/imu"/>
+    <remap from="/$(arg ouster_ns)/points" to="/os_cloud_node/points"/>
+    <remap from="/$(arg ouster_ns)/points2" to="/os_cloud_node/points2"/>
+
+Same thing needs to happen if we want to remap the topic names that are published by ``img_node``.
+This is shown below as well::
+
+    <remap from="/$(arg ouster_ns)/nearir_image" to="/img_node/nearir_image"/>
+    <remap from="/$(arg ouster_ns)/range_image" to="/img_node/range_image"/>
+    <remap from="/$(arg ouster_ns)/reflec_image" to="/img_node/reflec_image"/>
+    <remap from="/$(arg ouster_ns)/signal_image" to="/img_node/signal_image"/>
+
```

Comment:
Could be good to also mention that we dropped the default topic remapping for the bags recorded with [0b44a5](https://github.com/ouster-lidar/ouster_example/commit/0b44a5911d5c6b95d146cbdf9bd4d9695b045871) and earlier, and that they can still remap the `/lidar_packet` and `/imu_packet` topics as needed as well